src/lib.rs

```diff
@@ -84,12 +87,41 @@ impl<P> RefreshMessage<P> {
         P: ECPoint + Clone + Zeroize,
         P::Scalar: PartialEq + Clone + Debug + Zeroize,
     {
-        // TODO: make error verbose/output indices of malicious parties
         // check we got at least threshold t refresh messages
         if refresh_messages.len() <= old_key.t as usize {
-            return Err(());
+            return Err(FsDkrError::PartiesThresholdViolation {
+                threshold: old_key.t,
+                refreshed_keys: refresh_messages.len(),
+            });
+        }
+
+        // check all vectors are of same length
+        let reference_len = refresh_messages[0].fairness_proof_vec.len();
+
+        for k in 0..refresh_messages.len() {
+            let fairness_proof_len = refresh_messages[k].fairness_proof_vec.len();
+            let points_commited_len = refresh_messages[k].points_committed_vec.len();
+            let points_encrypted_len = refresh_messages[k].points_encrypted_vec.len();
+
+            if !(fairness_proof_len == reference_len
+                && points_commited_len == reference_len
+                && points_encrypted_len == reference_len)
+            {
+                return Err(FsDkrError::SizeMismatchError {
+                    refresh_message_index: k,
+                    fairness_proof_len,
+                    points_commited_len,
+                    points_encrypted_len,
+                });
+            }
+        }
+
+        // check that all the contents are not duplicated
```

Comment:
is this a real issue? if a duplicate message appears wouldn't it fail the proof and produce another error down the line?